src/pyxplor/pyxplor.py

```diff
@@ -0,0 +1,52 @@
+def time_series_explorer(input_df: pd.DataFrame, 
+                         date_column: str, 
+                         value_column: str, 
+                         freq: str = 'D', 
+                         plot_type: str = 'line', 
+                         figsize: tuple = (10, 6),
+                         output_path: str = None, 
+                         super_title: str = "Time Series Analysis",
+                         super_title_font: int = 14) -> None:
+    """
+    Conducts exploratory data analysis on time-series data and generates visualizations. 
+    This function plots key aspects of time-series data such as trends, seasonality, and patterns.
+
+    Parameters
+    ----------
+    input_df : pd.DataFrame
+        The DataFrame containing the time-series data.
+
+    date_column : str

```

Comment:
I think this should be a list? Otherwise, we would just be plotting one variable. 
eg. ["YearBuilt", "Date", "YearSold"]

Reply:
I agree!

Reply:
Also, in terms of `date_column` vs `value_column` - is `value_column` our target in the general case?

Reply:
Please check this version:

def xplor_time_series(input_df: pd.DataFrame, 
                      date_column: str, 
                      value_column: str, 
                      freq: str = 'D', 
                      figsize: tuple = (10, 6),
                      output_path: str = None, 
                      super_title: str = "Time Series Analysis",
                      super_title_font: int = 14) -> None:
    """
    Conducts exploratory data analysis on time-series data and generates line plot visualizations. 

    Parameters
    ----------
    input_df : pd.DataFrame
        The DataFrame containing the time-series data. Expected to have at least two columns: 
        one for dates and one for numeric values.
        
    date_column : str
        The name of the column in `input_df` that contains date/time information. 
        This column should be convertible to pandas datetime format.

    value_column : str
        The name of the column in `input_df` that contains the numeric values to be analyzed.

    freq : str, optional
        The frequency of the time-series data ('D' for daily, 'W' for weekly, etc.). Default is 'D'.

    figsize : tuple, optional
        The size of the plot in the format (width, height). Default is (10, 6).

    output_path : str, optional
        Path to save the plot. Defaults to None, which means the plot will not be saved.

    super_title : str, optional
        Super title for the entire plot. Default is "Time Series Analysis".

    super_title_font : int, optional
        Font size for the super title. Default is 14.

    Returns
    -------
    None
    """